### ISSUE TITLE
Blank lines not displayed in message reply

### ISSUE THREAD

**Issue by progfolio** [alphapapa/ement.el#252]
### OS/platform

Arch Linux

### Emacs version and provenance

GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.39, cairo version 1.18.0) of 2024-01-04

### Emacs command

emacs

### Emacs frame type

GUI

### Ement package version and provenance

version:  c8fa844 0.14-pre installed via Elpaca

### Actions taken

![ement-reply-formatting-bug](https://github.com/alphapapa/ement.el/assets/44036031/4a9cdc41-5a00-44b5-b6ec-6f9e749db7dc)

1. Send a message with blank lines. It is displayed properly.
2. Reply to the first message. The replay has blank lines in it. They are not displayed in the reply.
3. Note that when editing the message, the lines are preserved.

Element for Android display for comparison:

![image](https://github.com/alphapapa/ement.el/assets/44036031/79981d25-43bc-41be-83ce-832e85ca07a3)

### Observed results

The reply should be formatted as I wrote it.

### Expected results

The reply's formatting elides newlines.

**Comment by alphapapa** [comment]
Likely caused by using `shr` to render the reply inside an HTML `blockquote` tag.  Maybe we need to duplicate newlines, or replace them with `br` tags.  What do you think?

**Comment by progfolio** [comment]
Replacing newlines with "<br>" gets us part of the way there in `ement-room--render-html`:

```diff
-  (insert string))
+ (insert (replace-regexp-in-string "\n" "<br/>" string))
```

But that will also replace newlines in a reply's already formatted HTML, so it doesn't seem like the right solution to me.
Haven't had much time to dig into the source, but is there a way to separate the "in reply to" portion of a reply from the body of the actual reply?

**Comment by alphapapa** [comment]
> Haven't had much time to dig into the source, but is there a way to separate the quoted portion of a reply from the body of the actual reply?

Yes, but that is more easily handled by #150, which I want to merge soon, but haven't had time to test.  If anyone wants to test that and provide feedback, it would help.
